src/praisonai-agents/praisonaiagents/runtime/resolve.py

```
@@ -0,0 +1,378 @@
+_cleanup_interval = 600  # 10 minutes
+
+def _start_cleanup_thread():
+    """Start background cache cleanup thread."""
+    global _cleanup_thread
+    if _cleanup_thread is None or not _cleanup_thread.is_alive():
+        def cleanup_worker():
+            while True:
+                try:
+                    time.sleep(_cleanup_interval)
+                    _cleanup_expired_cache()
+                except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)